### ISSUE TITLE
restoring a model at gridsearchcv begins

### ISSUE THREAD

**Issue by andreapi87** [skorch-dev/skorch#855]
I want to fit a model from a previous saved session, but in a gridsearchCV.

I made the following callback to restore model params from file:
```
class RestoreModel(skorch.callbacks.Callback):
    TMP_PATH      = '/checkpoints'

    def __init__(self, parameters_filename,*wargs,
                 X_te=None, y_te=None,
                 **kwargs):
        super().__init__(*wargs, 
                         **kwargs)
        self.parameters_filename = parameters_filename
        self.X_te                = X_te
        self.y_te                = y_te
        
        
    def on_train_begin(self,net, **kwargs):        
        print(f"\nRestoring parameters from {self.parameters_filename}...", end='') 
        net.load_params(f_params=f'{RestoreModel.TMP_PATH}/{self.parameters_filename}')
        
        print("done!\n")
        if self.X_te is not None:
            import numpy
            print("testing...", end='')
            y_pred = net.predict(self.X_te)
            acc = numpy.mean(numpy.argmax(y_pred, axis=1)==self.y_te)
            print(f"accuracy of {acc}")
              
``` 

It works if it fit the model without cv, but if I use the model in GridSearchCV the saved weights are not used, starting the learning from scratch.

**Comment by andreapi87** [comment]
Ok, I think I solved setting `warm_start=True` in the net definition.

**Comment by andreapi87** [comment]
update: the problem remains also with `warm_start=True`

**Comment by BenjaminBossan** [comment]
Could you please share which hyper parameters you're grid-searching on?

**Comment by andreapi87** [comment]
> Could you please share which hyper parameters you're grid-searching on?

of course. Just learning rate, the other are fixed.

**Comment by BenjaminBossan** [comment]
Okay, that can't be the problem.

So I assume that with custom callback on top, when you grid search, the printed test score is not what it should be?

**Comment by andreapi87** [comment]
paradoxically, the printed test score is ok, but when the training starts the score crumbles. Instead, if I fit the model without passing for GridSearchCV, both the initial score and the loss are good.

**Comment by BenjaminBossan** [comment]
So let's take the net without grid search which seems to be working. Can you do:

```python
net.set_params(lr=lr)
net.fit(X, y)
```

with the same `lr` as you use in grid search? Are the results still good?

**Comment by andreapi87** [comment]
> So let's take the net without grid search which seems to be working. Can you do:
> 
> ```python
> net.set_params(lr=lr)
> net.fit(X, y)
> ```
> 
> with the same `lr` as you use in grid search? Are the results still good?

This is exactly what I do when I say that I fit the model without gridsearchv, and the results are good.

**Comment by BenjaminBossan** [comment]
I'm really not sure what the issue is. Could you try:

```python
from sklearn.base import clone
net2 = clone(net)
net2.set_params(lr=lr)
net2.fit(X, y)
```

**Comment by andreapi87** [comment]
I'm sorry. I've just realized that I inserted the wrong lr (0.1 instead of 0.01). Maybe this value leads my model toward a bad loss just at the first iteration, seeming that the model starts from the beginning. 
I'm sorry for my error, I close this non-existent issue.
